### DISCUSSION TITLE
Great, just great!

### DISCUSSION THREAD

**Discussion by thenicnic** [Pustur/whatsapp-chat-parser-website#14] [General]
Thank you very much for this absolutely brillant piece of software!
Saving the demo page is all you need for offline usage. Even the print mode isn't ink-consuming because of black-and-white bubbles.
One thing, a date filter as alternative to the line filter, would be cool. 

Absolutely outstanding project!

**Comment by Pustur** [comment]
Thanks a lot for the kind words, I'm glad you found the app useful!

You just reminded me that I should improve the print styles, by removing the header and floating button. I'll release a new version when I have time to improve that.

About the dates as filters I think that's a great idea, but I'm not sure how easy it would be to implement considering there would be 2 filtering methods at the same time. I'll have to think about it more but it's not a priority for now.

Thanks again and have a nice day.

**EDIT:** As a reminder, I provide download links in the [releases](https://github.com/Pustur/whatsapp-chat-parser-website/releases) for offline usage should you need them.

**Reply by thenicnic** [reply]
Thanks for your reply. I've already seen the packages from the Release page but the index file ony showed up as a blank page on my system so I tried it with just saving the existing web app.
The cosmetic optimizations to the print preview sound like a good idea. I'm not an experienced developer but you made me curious to have a look into the code to see if I can do anything for support.

I would judt integrate the filters as an XOR option. Combining lines AND date filters indeed seem to be hard to integrate and I cannot imagine a scenario where this is absolutely needed. 
(And in case it is, trimming the lines can be done before uploading to your web app and applying the date filter could be the next step). 

For me, two radio buttons to switch between line based and date based filtering would be fine.

**Reply by Pustur** [reply]
> Thanks for your reply. I've already seen the packages from the Release page but the index file ony showed up as a blank page on my system so I tried it with just saving the existing web app.

Oh you're right, that's because you still need a local server to see it properly as the file paths are absolute and don't work by just opening the index.html in the browser. Cool that you got it to work by just saving the page though!

> The cosmetic optimizations to the print preview sound like a good idea. I'm not an experienced developer but you made me curious to have a look into the code to see if I can do anything for support.

It's nothing too difficult, I plan to just use something like this:

```css
@media print {
  elements-that-should-not-be-printed {
    display: none;
  }
}
```

> For me, two radio buttons to switch between line based and date based filtering would be fine.

That seems like a reasonable idea. I think I'll go with this one to avoid complications. Will track this feature in #15
